lib/cn/keep.c

```diff
@@ -53,20 +73,46 @@ kvset_keep_vblocks(struct kvset_vblk_map *vbm, struct kv_iterator **iv, int niv)
      */
 
     nv = 0;
-    for (i = 0; i < niv; ++i) {
+    vgidx = 0;
+    for (int i = 0; i < niv; ++i) {
         struct kvset *kvset = kvset_from_iter(iv[i]);
-        int           cnt = kvset_get_num_vblocks(kvset);
+        int cnt = kvset_get_num_vblocks(kvset);
```

Comment:
We are not using the right type in several places and I think we should clean it up in a separate pass. We are saved here because the total number of vgroups cannot exceed 32K.

Reply:
Feels like in code you are touching in the PR, this can be corrected. Less code to correct elsewhere.